src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerWriteContextComplexJson.cs

```diff
@@ -326,6 +326,12 @@ private void VerifyType(DataContract dataContract, Type declaredType)
                 scopedKnownTypes.Pop();
             }
         }
+
+        internal static void WriteJsonNameWithMapping(XmlWriterDelegator xmlWriter, XmlDictionaryString[] memberNames, int index)
```

Comment:
I've added Debug.Assert to all applicable places that access to type members through reflection. I think there is validation check in NetNative toolchain but it shows as warning and not failure.